platform/posix/transport/src/openssl_posix.c

```diff
@@ -792,47 +827,70 @@ int32_t Openssl_Send( NetworkContext_t * pNetworkContext,
-
-            LogError( ( "Failed to send data over network: SSL_write of OpenSSL failed: "
-                        "ErrorStatus=%s.", ERR_reason_error_string( sslError ) ) );
-
-            /* As the SSL context is configured for blocking mode, the SSL_write() function
-             * does not return an SSL_ERROR_WANT_READ or SSL_ERROR_WANT_WRITE error code.
-             * The SSL_ERROR_WANT_READ and SSL_ERROR_WANT_WRITE error codes signify that
-             * the write operation can be retried.
-             * However, in the blocking mode, as the SSL_write() function does not return
-             * either of the error codes, we cannot retry the operation on failure, and thus,
-             * this function will never return a zero error code.
-             */
-
-            /* The transport interface requires zero return code only when the send operation can
-             * be retried to achieve success. Thus, convert a zero error code to a negative return
-             * value as this cannot be retried. */
-            if( bytesSent == 0 )
+            /* SSL write of data. */
+            bytesSent = ( int32_t ) SSL_write( pOpensslParams->pSsl, pBuffer,
+                                               ( int32_t ) bytesToSend );
+
+            if( bytesSent <= 0 )
             {
-                bytesSent = -1;
+                LogError(
+                    ( "Failed to send data over network: SSL_write of OpenSSL failed: "
+                      "ErrorStatus=%s.",
+                      ERR_reason_error_string( SSL_get_error( pOpensslParams->pSsl, bytesSent ) ) ) );
+
+                /* As the SSL context is configured for blocking mode, the SSL_write()
+                 * function does not return an SSL_ERROR_WANT_READ or
+                 * SSL_ERROR_WANT_WRITE error code. The SSL_ERROR_WANT_READ and
+                 * SSL_ERROR_WANT_WRITE error codes signify that the write operation can
+                 * be retried. However, in the blocking mode, as the SSL_write()
+                 * function does not return either of the error codes, we cannot retry
+                 * the operation on failure, and thus, this function will never return a
+                 * zero error code.
+                 */
+
+                /* The transport interface requires zero return code only when the send
+                 * operation can be retried to achieve success. Thus, convert a zero
+                 * error code to a negative return value as this cannot be retried. */
+                if( bytesSent == 0 )
+                {
+                    bytesSent = -1;
+                }
             }
         }
+        else if( pollStatus < 0 )
+        {
+            /* An error occurred while polling. */
+            LogError( ( "An error occurred while polling." ) );
+            bytesSent = -1;
+        }
+        else
+        {
+            /* Timed out waiting for data to be sent. */
```

Comment:
There was no timed wait for the socket to be available as the `timeout` parameter used is `0`. So, this comment needs to be updated
```suggestion
            /* Socket is not available for sending data. Set return code for retrying send. */
```